controllers/node_controller.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/openshift/windows-machine-config-operator/pkg/condition"
+	"github.com/openshift/windows-machine-config-operator/pkg/metadata"
+	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig"
+	"github.com/openshift/windows-machine-config-operator/pkg/secrets"
+	"github.com/openshift/windows-machine-config-operator/pkg/signer"
+)
+
+const (
+	// NodeController is the name of this controller in logs and other outputs.
+	NodeController = "node"
+)
+
+// nodeReconciler holds the info required to reconcile a Node object, inclduing that of the underlying Windows instance
+type nodeReconciler struct {
+	instanceReconciler
+}
+
+// NewNodeReconciler returns a pointer to a new nodeReconciler
+func NewNodeReconciler(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) (*nodeReconciler, error) {
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return nil, fmt.Errorf("error creating kubernetes clientset: %w", err)
+	}
+
+	return &nodeReconciler{
+		instanceReconciler: instanceReconciler{
+			client:             mgr.GetClient(),
+			log:                ctrl.Log.WithName("controllers").WithName(NodeController),
+			k8sclientset:       clientset,
+			clusterServiceCIDR: clusterConfig.Network().GetServiceCIDR(),
+			watchNamespace:     watchNamespace,
+			recorder:           mgr.GetEventRecorderFor(NodeController),
+		},
+	}, nil
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which reads that state of the cluster for a
+// CertificateSigningRequests object and aims to move the current state of the cluster closer to the desired state.
+func (r *nodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	_ = r.log.WithValues(NodeController, req.NamespacedName)
+	// Prevent WMCO upgrades while Node objects are being processed
+	if err := condition.MarkAsBusy(r.client, r.watchNamespace, r.recorder, NodeController); err != nil {
+		return ctrl.Result{}, err
+	}
+	defer func() {
+		err = markAsFreeOnSuccess(r.client, r.watchNamespace, r.recorder, NodeController, result.Requeue, err)
+	}()
+
+	// Fetch Node reference
+	node := &core.Node{}
+	if err := r.client.Get(ctx, req.NamespacedName, node); err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - return error to requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Create a new signer using the private key that the instances will be configured with
+	r.signer, err = signer.Create(types.NamespacedName{Namespace: r.watchNamespace,
+		Name: secrets.PrivateKeySecret}, r.client)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to create signer from private key secret: %w", err)
+	}
```

Comment:
Setting the signer like this is a bit weird.
Instead you can just assign it to a new variable and use it directly in the NewNodeConfig() call